src/issues/currentIssue.ts

```diff
@@ -92,34 +95,43 @@ export class CurrentIssue {
 		}
 	}
 
+	private async getUser(): Promise<string> {
+		if (!this.user) {
+			this.user = await this.issueModel.githubRepository.getAuthenticatedUser();
+		}
+		return this.user;
+	}
+
 	private async createIssueBranch(): Promise<void> {
 		const createBranchConfig = this.shouldPromptForBranch ? 'prompt' : <string>vscode.workspace.getConfiguration(ISSUES_CONFIGURATION).get(BRANCH_CONFIGURATION);
 		if (createBranchConfig === 'off') {
 			return;
 		}
 		const state: IssueState = this.stateManager.getSavedIssueState(this.issueModel.number);
 		this._branchName = this.shouldPromptForBranch ? undefined : state.branch;
-		let user: string | undefined;
 		if (!this._branchName) {
-			user = await this.issueModel.githubRepository.getAuthenticatedUser();
 			if (createBranchConfig === 'on') {
 				const branchNameConfig = <string>vscode.workspace.getConfiguration(ISSUES_CONFIGURATION).get(BRANCH_NAME_CONFIGURATION);
-				this._branchName = await variableSubstitution(branchNameConfig, this.issue, undefined, user);
+				this._branchName = await variableSubstitution(branchNameConfig, this.issue, undefined, await this.getUser());
 			} else {
 				this._branchName = await vscode.window.showInputBox({ placeHolder: `issue${this.issueModel.number}`, prompt: 'Enter the label for the new branch.' });
 			}
 		}
 		if (!this._branchName) {
-			user = await this.issueModel.githubRepository.getAuthenticatedUser();
-			this._branchName = this.getBasicBranchName(user);
+			this._branchName = this.getBasicBranchName(await this.getUser());
 		}
 
 		state.branch = this._branchName;
 		this.stateManager.setSavedIssueState(this.issueModel, state);
 		try {
 			await this.createOrCheckoutBranch(this._branchName);
 		} catch (e) {
-			const basicBranchName = this.getBasicBranchName(user ?? await this.issueModel.githubRepository.getAuthenticatedUser());
+			const basicBranchName = this.getBasicBranchName(await this.getUser() ?? await this.issueModel.githubRepository.getAuthenticatedUser());
```

Comment:
aren't these two statements equivalent?

Reply:
Yes, bad refactoring. Thanks for catching!